In [8]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import statsmodels.api as sm
import statsmodels.formula.api as smf
%matplotlib inline
from sklearn import datasets, linear_model
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split

In [9]:
df_salary = pd.read_csv("salary.csv").dropna()
df_salary.head()

salary  exprior  yearsworked  yearsrank  market  degree  otherqual  \
0  53000.0        0            0          0    1.17       1          0   
1  58000.0        1            0          0    1.24       1          0   
2  45500.0        0            0          0    1.21       1          0   
3  35782.0        0            2          1    0.99       1          0   
4  34731.0        0            2          2    0.91       1          0   

   position  male  Field  yearsabs  
0         1     1      3         0  
1         1     1      2         0  
2         1     1      3         2  
3         1     1      4         1  
4         1     1      4         1

In [10]:
df_salary.info()
df_salary.shape

<class 'pandas.core.frame.DataFrame'>
Int64Index: 513 entries, 0 to 513
Data columns (total 11 columns):
salary         513 non-null float64
exprior        513 non-null int64
yearsworked    513 non-null int64
yearsrank      513 non-null int64
market         513 non-null float64
degree         513 non-null int64
otherqual      513 non-null int64
position       513 non-null int64
male           513 non-null int64
Field          513 non-null int64
yearsabs       513 non-null int64
dtypes: float64(2), int64(9)
memory usage: 48.1 KB


(513, 11)

In [24]:

y = df_salary.salary

# create training and testing vars
x_train, x_test, y_train, y_test = train_test_split(df_salary, y, test_size=0.3, random_state = 15)
print(x_train.shape, y_train.shape)
print(x_test.shape, y_test.shape)

# create a fitted model in one line
lm = smf.ols(formula='salary ~ yearsworked', data=x_train).fit()

# print the coefficients
print(lm.params)
print(lm.summary())

(359, 11) (359,)
(154, 11) (154,)
Intercept      39296.664658
yearsworked      891.581770
dtype: float64
                            OLS Regression Results                            
Dep. Variable:                 salary   R-squared:                       0.448
Model:                            OLS   Adj. R-squared:                  0.447
Method:                 Least Squares   F-statistic:                     289.9
Date:                Fri, 17 May 2019   Prob (F-statistic):           5.22e-48
Time:                        14:23:36   Log-Likelihood:                -3793.9
No. Observations:                 359   AIC:                             7592.
Df Residuals:                     357   BIC:                             7600.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------

### Q1-
#### a. The model does predict the dependent variable which is salary, with the r squared being 0.447 the points are a bit far apart indidcating that it is slightly unfit and the p-value being 0.000 it is considered to be statistically significant.
#### b. The percentage of the variance in employees’ salaries is 44.7%.

### Q2- What does the unstandardized coefficient (B or 'coef' in statsmodels) tell you about the relationship between Years Worked and Salary?
#### 